vars/MavenCommand.groovy

```diff
@@ -0,0 +1,127 @@
+    List profiles = []
+
+    MavenCommand(steps){
+        this.steps = steps
+    }
+
+    MavenCommand(steps, List defaultOpts){
+        this(steps)
+        this.mavenOptions.addAll(defaultOpts)
+    }
+
+    def run(String goals) {
+        def cmd = "mvn -B"
+        if(this.settingsXmlPath) {
+            cmd += " -s ${this.settingsXmlPath}"
+        }
+        if(this.mavenOptions.size() > 0){
+            cmd += " ${this.mavenOptions.join(' ')}"
+        }
+        cmd += " ${goals}"
+        if(this.profiles.size() > 0){
+            cmd += " -P${this.profiles.join(',')}"
+        }
+        if(this.properties.size()){
+            cmd += " ${this.properties.collect{ it.value != '' ? "-D${it.key}=${it.value}" : "-D${it.key}" }.join(' ')}"
+        }
+        if(this.logFileName){
+            cmd += " | tee \$WORKSPACE/${logFileName} ; test \${PIPESTATUS[0]} -eq 0"
+        }
+
+        if(this.returnStdout){
+            return steps.sh(script: cmd, returnStdout: this.returnStdout)
+        }
+        steps.sh cmd
+    }
+
+    MavenCommand withSettingsXmlId(String settingsXmlId){
+        steps.configFileProvider([steps.configFile(fileId: settingsXmlId, variable: 'MAVEN_SETTINGS_XML')]) {
+            withSettingsXmlFile(steps.env['MAVEN_SETTINGS_XML'])
+        }
+        return this
+    }
+
+    MavenCommand withSettingsXmlFile(String settingsXmlPath){
+        if(!settingsXmlPath){
+            error 'Trying to set an empty settings xml path'
+        }
+        this.settingsXmlPath = settingsXmlPath
+        return this
+    }
+
+    MavenCommand withOptions(List opts) {
+        this.mavenOptions.addAll(opts)
+        return this
+    }
+
+    MavenCommand skipTests(boolean skipTests=true) {
+        return withProperty('skipTests', skipTests)
+    }
+
+    MavenCommand withProfiles(List profiles) {
+        this.profiles.addAll(profiles)
+        return this
+    }
+
+    MavenCommand withProperty(String key, Object value = '') {
+        this.properties.put(key, value)
+        return this
+    }
+
+    MavenCommand withProperties(Properties properties) {
+        properties.each { this.properties.put(it.key, it.value) }
+        return this
+    }
+
+    MavenCommand withPropertyMap(Map properties) {
+        this.properties.putAll(properties)
+        return this
+    }
+
+    MavenCommand withLogFileName(String logFileName){
+        if(!logFileName){
+            error 'Trying to set an empty log file name'
+        }
+        this.logFileName = logFileName
+        return this
+    }
+
+    MavenCommand withDeployRepository(String deployRepository){
+        if(!deployRepository){
+            error 'Trying to add an empty deploy repository'
+        }
+        withProperty('altDeploymentRepository', "runtimes-artifacts::default::${deployRepository}")
+        withProperty('enforcer.skip', true)
+    }
+
+    MavenCommand withLocalDeployFolder(String localDeployFolder){
+        if(!localDeployFolder){
```

Comment:
I would replace this by
`assert localDeployFolder: 'Trying to add an empty local deploy folder'`